src/cnxml/util.js

```diff
@@ -361,6 +361,42 @@ export function loadClasses(el) {
     return List(classes.trim().split(/\s+/))
 }
 
+/**
+ * Return element's attributes as an object.
+ */
+export function loadAttributes(el) {
+    const attrs = {}
+
+    for (const attr of el.attributes) {
+        if (attr.name === 'mime-type') {
+            attrs.mime = attr.value
+            continue
+        }
+        attrs[attr.name] = attr.value
+    }
+
+    return attrs
+}
+
+/**
+ * Convert Slate~Data to attributes for cnxml node element.
+ * @param {Slate~Data} data
+ */
+export function convertDataToAttributes(data) {
+    const attrs = {}
+    for (const [attr, val] of data.entries()) {
+        if (attr === 'class') {
+            attrs[attr] = val.join(' ')
+            continue
+        } else if (attr === 'mime') {
+            attrs['mime-type'] = val
+            continue
+        }
+        attrs[attr] = val
+    }
+    return attrs
+}
```

Comment:
I'm not convinced we should indiscriminately load all attributes, and attributes should be loaded as their proper types, not as strings.

Reply:
I've added this to not repeat myself when reading attrs for media files and in the future for elements related to tables since there will be many of the same attrs.
I agree that elements should be loaded as proper types, not strings.

What is your suggestion for handling this overall?
- Duplicate converting attrs for each media file
- Leave it as it is + convert attrs to proper types + omit unwanted attrs?
- Something else?